pkg/patterns/declarative/reconciler.go

```diff
@@ -446,3 +493,41 @@ func parseListKind(infos *manifest.Objects) (*manifest.Objects, error) {
 
 	return &ret, nil
 }
+
+
+func aggregateStatus(m map[status.Status]bool) string {
+	inProgress := m["InProgress"]
+	terminating := m["Terminating"]
+
+	failed := m["Failed"]
+
+	if inProgress || terminating {
+		return "InProgress"
+	}
+
+	if failed {
+		return "Failed"
+	}
+	return "Current"
+}
+
+func getObjectFromCluster(obj *manifest.Object, config *rest.Config, dynamicClient dynamic.Interface) (*unstructured.
+	Unstructured, error) {
+	getOptions := metav1.GetOptions{}
+	gvk := obj.GroupVersionKind()
+	restMapper, err := apiutil.NewDiscoveryRESTMapper(config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create RESTMapper from config: %v", err)
+	}
+	mapping, err := restMapper.RESTMapping(obj.GroupKind(), gvk.Version)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get mapping for resource: %v", err)
+	}
+	ns := obj.UnstructuredObject().GetNamespace()
+	unstruct, err := dynamicClient.Resource(mapping.Resource).Namespace(ns).Get(context.Background(),
```

Comment:
Nit: always better to pass in the context if you control the method signature... even if the caller just specifies `context.TODO()`, it just avoids having to refactor in future.

Reply:
I am not completely sure I understand